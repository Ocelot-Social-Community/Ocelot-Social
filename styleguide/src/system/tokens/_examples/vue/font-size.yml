#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
