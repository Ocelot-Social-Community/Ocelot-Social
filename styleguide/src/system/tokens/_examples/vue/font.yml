#
# FONT TOKENS
# Use these tokens to set font-weight and font-family.
#

props:
  - name: font-family-heading
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-text
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-code
    value: "inconsolata, monospace"
    category: "font-family"

  - name: font-weight-regular
    value: "normal"
    category: "font-weight"
  - name: font-weight-bold
    value: "600"
    category: "font-weight"
global:
  type: ...
  category: font
