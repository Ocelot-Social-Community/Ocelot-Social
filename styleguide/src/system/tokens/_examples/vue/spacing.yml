#
# SPACING TOKENS
# Use these tokens to set padding, margin and position coordinates.
#

aliases:
  xxx-large:
    value: &xxx-large 120
  xx-large:
    value: &xx-large 60
  x-large:
    value: &x-large 40
  large:
    value: &large 28
  base:
    value: &base 20
  small:
    value: &small 12
  x-small:
    value: &x-small 6
  xx-small:
    value: &xx-small 4
  xxx-small:
    value: &xxx-small 2

props:
  - name: xxx-large
    value: *xxx-large
    category: space-size
  - name: xx-large
    value: *xx-large
    category: space-size
  - name: x-large
    value: *x-large
    category: space-size
  - name: large
    value: *large
    category: space-size
  - name: base
    value: *base
    category: space-size
  - name: small
    value: *small
    category: space-size
  - name: x-small
    value: *x-small
    category: space-size
  - name: xx-small
    value: *xx-small
    category: space-size
  - name: xxx-small
    value: *xxx-small
    category: space-size

  - name: space-xxx-large
    value: "{!xxx-large}px"
  - name: space-xx-large
    value: "{!xx-large}px"
  - name: space-x-large
    value: "{!x-large}px"
  - name: space-large
    value: "{!large}px"
  - name: space-base
    value: "{!base}px"
  - name: space-small
    value: "{!small}px"
  - name: space-x-small
    value: "{!x-small}px"
  - name: space-xx-small
    value: "{!xx-small}px"
  - name: space-xxx-small
    value: "{!xxx-small}px"

global:
  type: number
  category: space
