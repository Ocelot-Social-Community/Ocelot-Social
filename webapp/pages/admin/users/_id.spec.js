import { render, fireEvent, screen } from '@testing-library/vue'
import BadgesPage from './_id.vue'

const localVue = global.localVue

const availableBadges = [
  {
    id: 'verification-badge-1',
    icon: 'icon1',
    type: 'verification',
    description: 'description-v-1',
  },
  {
    id: 'verification-badge-2',
    icon: 'icon2',
    type: 'verification',
    description: 'description-v-2',
  },
  {
    id: 'trophy-badge-1',
    icon: 'icon3',
    type: 'trophy',
    description: 'description-t-1',
  },
  {
    id: 'trophy-badge-2',
    icon: 'icon4',
    type: 'trophy',
    description: 'description-t-2',
  },
]

const user = {
  id: 'user1',
  verified: {
    id: 'verification-badge-1',
  },
  badges: [
    {
      id: 'trophy-badge-2',
    },
  ],
}

describe('.vue', () => {
  let wrapper
  let mocks

  beforeEach(() => {
    mocks = {
      $t: jest.fn((v) => v),
      $apollo: {
        User: {
          query: jest.fn(),
        },
        badges: {
          query: jest.fn(),
        },
        mutate: jest.fn(),
      },
      $toast: {
        success: jest.fn(),
        error: jest.fn(),
      },
    }
  })
  const Wrapper = () => {
    return render(BadgesPage, {
      mocks,
      localVue,
      data: () => ({
        user,
        badges: availableBadges,
      }),
    })
  }

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('renders', () => {
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  describe('after clicking an inactive verification badge', () => {
    let button
    beforeEach(() => {
      button = screen.getByAltText(availableBadges[1].description)
    })

    describe('and successful server response', () => {
      beforeEach(async () => {
        mocks.$apollo.mutate.mockResolvedValue({
          data: {
            setVerificationBadge: {
              id: 'user1',
              verified: {
                id: availableBadges[1].id,
              },
              badges: [],
            },
          },
        })
        await fireEvent.click(button)
      })

      it('calls the mutation', async () => {
        expect(mocks.$apollo.mutate).toHaveBeenCalledWith({
          mutation: expect.anything(),
          variables: {
            badgeId: availableBadges[0].id,
            userId: 'user1',
          },
        })
      })

      it('shows success message', async () => {
        expect(mocks.$toast.success).toHaveBeenCalledWith('admin.badges.setVerification.success')
      })
    })

    describe('and failed server response', () => {
      beforeEach(async () => {
        mocks.$apollo.mutate.mockRejectedValue({ message: 'Ouch!' })
        await fireEvent.click(button)
      })

      it('calls the mutation', async () => {
        expect(mocks.$apollo.mutate).toHaveBeenCalledWith({
          mutation: expect.anything(),
          variables: {
            badgeId: availableBadges[0].id,
            userId: 'user1',
          },
        })
      })

      it('shows error message', async () => {
        expect(mocks.$toast.error).toHaveBeenCalledWith('admin.badges.setVerification.error')
      })
    })

    describe('after clicking an inactive trophy badge', () => {
      let button
      beforeEach(() => {
        button = screen.getByAltText(availableBadges[2].description)
      })

      describe('and successful server response', () => {
        beforeEach(async () => {
          mocks.$apollo.mutate.mockResolvedValue({
            data: {
              setTrophyBadge: {
                id: 'user1',
                verified: null,
                badges: [
                  {
                    id: 'trophy-badge-2',
                  },
                ],
              },
            },
          })
          await fireEvent.click(button)
        })

        it('calls the mutation', async () => {
          expect(mocks.$apollo.mutate).toHaveBeenCalledWith({
            mutation: expect.anything(),
            variables: {
              badgeId: availableBadges[3].id,
              userId: 'user1',
            },
          })
        })

        it('shows success message', async () => {
          expect(mocks.$toast.success).toHaveBeenCalledWith('admin.badges.rewardTrophy.success')
        })
      })

      describe('and failed server response', () => {
        beforeEach(async () => {
          mocks.$apollo.mutate.mockRejectedValue({ message: 'Ouch!' })
          await fireEvent.click(button)
        })

        it('calls the mutation', async () => {
          expect(mocks.$apollo.mutate).toHaveBeenCalledWith({
            mutation: expect.anything(),
            variables: {
              badgeId: availableBadges[3].id,
              userId: 'user1',
            },
          })
        })

        it('shows error message', async () => {
          expect(mocks.$toast.error).toHaveBeenCalledWith('admin.badges.rewardTrophy.error')
        })
      })
    })

    describe('after clicking an active verification badge', () => {
      let button
      beforeEach(() => {
        button = screen.getByAltText(availableBadges[0].description)
      })

      describe('and successful server response', () => {
        beforeEach(async () => {
          mocks.$apollo.mutate.mockResolvedValue({
            data: {
              setVerificationBadge: {
                id: 'user1',
                verified: null,
                badges: [],
              },
            },
          })
          await fireEvent.click(button)
        })

        it('calls the mutation', async () => {
          expect(mocks.$apollo.mutate).toHaveBeenCalledWith({
            mutation: expect.anything(),
            variables: {
              badgeId: availableBadges[0].id,
              userId: 'user1',
            },
          })
        })

        it('shows success message', async () => {
          expect(mocks.$toast.success).toHaveBeenCalledWith('admin.badges.revoke.success')
        })
      })

      describe('and failed server response', () => {
        beforeEach(async () => {
          mocks.$apollo.mutate.mockRejectedValue({ message: 'Ouch!' })
          await fireEvent.click(button)
        })

        it('calls the mutation', async () => {
          expect(mocks.$apollo.mutate).toHaveBeenCalledWith({
            mutation: expect.anything(),
            variables: {
              badgeId: availableBadges[0].id,
              userId: 'user1',
            },
          })
        })

        it('shows error message', async () => {
          expect(mocks.$toast.error).toHaveBeenCalledWith('admin.badges.revoke.error')
        })
      })
    })
  })

  describe('after clicking an active trophy badge', () => {
    let button
    beforeEach(() => {
      button = screen.getByAltText(availableBadges[3].description)
    })

    describe('and successful server response', () => {
      beforeEach(async () => {
        mocks.$apollo.mutate.mockResolvedValue({
          data: {
            setTrophyBadge: {
              id: 'user1',
              verified: null,
              badges: [],
            },
          },
        })
        await fireEvent.click(button)
      })

      it('calls the mutation', async () => {
        expect(mocks.$apollo.mutate).toHaveBeenCalledWith({
          mutation: expect.anything(),
          variables: {
            badgeId: availableBadges[3].id,
            userId: 'user1',
          },
        })
      })

      it('shows success message', async () => {
        expect(mocks.$toast.success).toHaveBeenCalledWith('admin.badges.revoke.success')
      })
    })

    describe('and failed server response', () => {
      beforeEach(async () => {
        mocks.$apollo.mutate.mockRejectedValue({ message: 'Ouch!' })
        await fireEvent.click(button)
      })

      it('calls the mutation', async () => {
        expect(mocks.$apollo.mutate).toHaveBeenCalledWith({
          mutation: expect.anything(),
          variables: {
            badgeId: availableBadges[3].id,
            userId: 'user1',
          },
        })
      })

      it('shows error message', async () => {
        expect(mocks.$toast.error).toHaveBeenCalledWith('admin.badges.revoke.error')
      })
    })
  })
})
