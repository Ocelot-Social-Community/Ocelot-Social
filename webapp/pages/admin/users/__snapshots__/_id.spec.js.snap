// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.vue renders 1`] = `
<body>
  <div>
    <section
      class="ds-section"
    >
      <div
        class="ds-section-content"
      >
        <div
          class="ds-container ds-container-x-large"
        >
          <div
            class="ds-space"
            style="margin-bottom: 32px;"
          >
            <h1
              class="ds-heading ds-heading-h3"
            >
              admin.badges.title
            </h1>
             
            <p
              class="ds-text"
            >
              admin.badges.description
            </p>
          </div>
           
          <article
            class="base-card"
          >
            <div
              class="badge-section"
            >
              <h4>
                admin.badges.verificationBadges
              </h4>
               
              <div
                class="badge-container"
              >
                <button
                  class="badge"
                >
                  <img
                    alt="description-v-1"
                    src="icon1"
                  />
                </button>
                <button
                  class="badge inactive"
                >
                  <img
                    alt="description-v-2"
                    src="icon2"
                  />
                </button>
              </div>
            </div>
             
            <div
              class="badge-section"
            >
              <h4>
                admin.badges.trophyBadges
              </h4>
               
              <div
                class="badge-container"
              >
                <button
                  class="badge inactive"
                >
                  <img
                    alt="description-t-1"
                    src="icon3"
                  />
                </button>
                <button
                  class="badge"
                >
                  <img
                    alt="description-t-2"
                    src="icon4"
                  />
                </button>
              </div>
            </div>
             
            <!---->
          </article>
        </div>
      </div>
    </section>
  </div>
</body>
`;
